AEPMessaging/Sources/Message.swift

```diff
@@ -146,6 +128,44 @@ public class Message: NSObject {
         if autoTrack {
             track(nil, withEdgeEventType: .trigger)
         }
+        recordEventHistory(eventType: .trigger, interaction: nil)
+    }
+    
+    /// Dispatches an event to be recorded in Event History.
+    ///
+    /// Record is created using the `propositionInfo.activityId` for this message.
+    ///
+    /// - Parameters:
+    ///  - eventType: `MessagingEdgeEventType` to be recorded
+    ///  - interaction: if provided, adds a custom interaction to the hash
+    func recordEventHistory(eventType: MessagingEdgeEventType, interaction: String?) {
+        guard let propInfo = propositionInfo else {
+            Log.debug(label: MessagingConstants.LOG_TAG, "Unable to write event history event '\(eventType.propositionEventType)', proposition info is not available for message (\(id)).")
+            return
+        }
+        
+        // iam dictionary used for event history
+        let iamHistory: [String: String] = [
+            MessagingConstants.Event.History.Keys.EVENT_TYPE: eventType.propositionEventType,
+            MessagingConstants.Event.History.Keys.MESSAGE_ID: propInfo.activityId,
+            MessagingConstants.Event.History.Keys.TRACKING_ACTION: interaction ?? ""
+        ]
+
+        // wrap history in an "iam" object
+        let eventHistoryData: [String: Any] = [
+            MessagingConstants.Event.Data.Key.IAM_HISTORY: iamHistory
+        ]
+        
+        let mask = [
+            MessagingConstants.Event.History.Mask.EVENT_TYPE,
+            MessagingConstants.Event.History.Mask.MESSAGE_ID,
+            MessagingConstants.Event.History.Mask.TRACKING_ACTION
+        ]
+        
+        let interactionLog = interaction == nil ? "" : " with value '\(interaction ?? "")'"
+        Log.trace(label: MessagingConstants.LOG_TAG, "Writing '\(eventType.propositionEventType)' event\(interactionLog) to EventHistory for in-app message with activityId '\(propInfo.activityId)'")
```

Comment:
nit: Can we not just log the info like below?
 ```
 Log.trace(label: MessagingConstants.LOG_TAG, "Writing '\(eventType.propositionEventType)' event with value \(interaction ?? "") to EventHistory for in-app message with activityId '\(propInfo.activityId)'")
 ```
 Or are you trying to discern if empty string is supplied for interaction vs interaction is nil?

Reply:
there is only a value for `interact` event types. so it looks like the following for each of our 4 types: 
```
Writing 'trigger' event to EventHistory for activityId 'd2b0e969-0378-4878-b669-6741cca0c770#b4ae863b-e517-42da-9af7-897adeff1157'
Writing 'display' event to EventHistory for activityId 'd2b0e969-0378-4878-b669-6741cca0c770#b4ae863b-e517-42da-9af7-897adeff1157'
Writing 'interact' event with value 'clicked' to EventHistory for activityId 'd2b0e969-0378-4878-b669-6741cca0c770#b4ae863b-e517-42da-9af7-897adeff1157'
Writing 'dismiss' event to EventHistory for activityId 'd2b0e969-0378-4878-b669-6741cca0c770#b4ae863b-e517-42da-9af7-897adeff1157'
```